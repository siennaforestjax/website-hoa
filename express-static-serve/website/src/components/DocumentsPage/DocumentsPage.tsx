import React, { useEffect, useState } from 'react';
import classes from './DocumentsPage.module.css';
import HoaDocument from '../../interfaces/HoaDocument';
import sortStringDates from '../../helper-functions/sortStringDates';
import Document from '../Document/Document';
import axios from 'axios';

const API_URL = 'https://siennaforestjax.azurewebsites.net/'; //http://localhost:5000

function DocumentsPage() {
  const [docsMinutes, setDocsMinutes] = useState([]);
  const [docsMisc, setDocsMisc] = useState([]);
  const [isLoadingMinutes, setIsLoadingMinutes] = useState(true);
  const [isLoadingMisc, setIsLoadingMisc] = useState(true);
  const [minutesDidError, setMinutesDidError] = useState(false);
  const [miscDidError, setMiscDidError] = useState(false);
  
  useEffect(() => {
    axios.get(`${API_URL}/api/documents/minutes`)
        .then(response => {
          setDocsMinutes(response.data);
          setIsLoadingMinutes(false);
        })
        .catch(err => {
          setIsLoadingMinutes(false);
          setMinutesDidError(true);
        });
  }, [])
  useEffect(() => {
    axios.get(`${API_URL}/api/documents/misc`)
        .then(response => {
          setDocsMisc(response.data);
          setIsLoadingMisc(false);
        })
        .catch(err => {
          setIsLoadingMisc(false);
          setMiscDidError(true);
        });
  }, [])

  function displayDocuments(docs: Array<HoaDocument>) {
    console.log(`sorting ${docs.length} docs`);
    return (
      <div className={classes.fileContainer}>
        <ul className={classes.fileList}>
          {docs
            .sort((x, y) => sortStringDates(x.createTime, y.createTime))
            .map((doc) => (
              <Document key={doc.filename} document={doc} />
            ))}
        </ul>
      </div>
    );
  }

  return (
    <div className={'pageWrapper ' + classes.pageWrapperOverride}>
      <h1>Documents</h1>
      <h2>HOA Meeting Notes</h2>
      {
        isLoadingMinutes ? <p>Loading . . .</p> 
        : minutesDidError ? <p>Error!</p> : displayDocuments(docsMinutes)}

      <h2>Community Documents</h2>
      {
        isLoadingMisc ? <p>Loading . . .</p> :
        miscDidError ? <p>Error!</p> : displayDocuments(docsMisc)
      }
    </div>
  );
}

export default DocumentsPage;
