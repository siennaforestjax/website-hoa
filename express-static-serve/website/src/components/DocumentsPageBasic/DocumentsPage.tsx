import React, { useEffect, useState } from 'react';
import classes from './DocumentsPage.module.css';
import HoaDocument from '../../interfaces/HoaDocument';
import sortStringDates from '../../helper-functions/sortStringDates';
import Document from '../DocumentBasic/Document';
import axios from 'axios';
import LoadingSvg from '../LoadingSvg/LoadingSvg';

const API_URL = process.env.REACT_APP_API_URL;

function DocumentsPagePretty() {
  const [docsMinutes, setDocsMinutes] = useState([]);
  const [docsMisc, setDocsMisc] = useState([]);
  const [docsEssential, setDocsEssential] = useState([]);
  const [isLoadingMinutes, setIsLoadingMinutes] = useState(true);
  const [isLoadingMisc, setIsLoadingMisc] = useState(true);
  const [minutesDidError, setMinutesDidError] = useState(false);
  const [miscDidError, setMiscDidError] = useState(false);

  useEffect(() => {
    axios
      .get(`${API_URL}/api/documents/minutes`)
      .then((response) => {
        setDocsMinutes(response.data);
        setIsLoadingMinutes(false);
      })
      .catch((err) => {
        setIsLoadingMinutes(false);
        setMinutesDidError(true);
      });
  }, []);
  useEffect(() => {
    axios
      .get(`${API_URL}/api/documents/misc`)
      .then((response) => {
        setDocsMisc(response.data);
        setIsLoadingMisc(false);
      })
      .catch((err) => {
        setIsLoadingMisc(false);
        setMiscDidError(true);
      });
  }, []);
  useEffect(() => {
    axios
      .get(`${API_URL}/api/documents/essential`)
      .then((response) => {
        setDocsEssential(response.data);
        setIsLoadingEssential(false);
      })
      .catch((err) => {
        setIsLoadingEssential(false);
        setEssentialDidError(true);
      });
  }, []);

  function displayDocuments(docs: Array<HoaDocument>) {
    return (
      <ul className={classes.fileList}>
        {docs
          .sort((x, y) => sortStringDates(x.createTime, y.createTime))
          .map((doc) => (
            <Document key={doc.filename} document={doc} />
          ))}
      </ul>
    );
  }

  return (
    <>
      <h1>Documents</h1>
      <div className={classes.fileContainer}>
        <h2 className={classes.sectionTitle}>Homeowner Essentials</h2>
        {isLoadingMinutes ? (
          <LoadingSvg scale={0.8} />
        ) : minutesDidError ? (
          <p>Error Gathering Documents</p>
        ) : (
          displayDocuments(docsEssential)
        )}
      </div>
      <div className={classes.fileContainer}>
        <h2 className={classes.sectionTitle}>HOA Meeting Notes</h2>
        {isLoadingMinutes ? (
          <LoadingSvg scale={0.8} />
        ) : minutesDidError ? (
          <p>Error Gathering Documents</p>
        ) : (
          displayDocuments(docsMinutes)
        )}
      </div>
      <div className={classes.fileContainer}>
        <h2 className={classes.sectionTitle}>Miscellaneous Documents</h2>
        {isLoadingMisc ? (
          <LoadingSvg scale={0.8} />
        ) : miscDidError ? (
          <p>Error Gathering Documents</p>
        ) : (
          displayDocuments(docsMisc)
        )}
      </div>
    </>
  );
}

export default DocumentsPagePretty;
