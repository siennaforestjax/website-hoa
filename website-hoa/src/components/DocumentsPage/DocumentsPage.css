h2 {
    font-size: 1.2rem;
}

h2:first-of-type {
    margin-top: 15px;
}

.file-container li {
    padding-left: calc(var(--gutter)*2);
}

.file-container {
    display: block;
    margin-bottom: 25px;
}

