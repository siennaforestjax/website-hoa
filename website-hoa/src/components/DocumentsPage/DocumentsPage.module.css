.pageWrapperOverride {
  align-items: flex-start;
}

.docList {
  list-style-type: none;
}

h2 {
  font-size: 1.2rem;
}

h2:first-of-type {
  margin-top: 15px;
}

.fileContainer li {
  padding-left: calc(var(--gutter) * 2);
}

.fileContainer {
  display: block;
  margin-bottom: 25px;
}

.doc {
  padding: 5px;
}

/*--------------Medium Screens--------*/
@media screen and (min-width: 768px) {
  .docList {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    width: 80%;
    justify-content: center;
  }

  .doc {
    display: block;
    width: 50%;
    padding: 10px;
  }
}
