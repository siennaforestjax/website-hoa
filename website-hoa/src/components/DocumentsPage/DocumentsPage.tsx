import React from 'react';
import classes from './DocumentsPage.module.css';
import HoaDocument from '../../interfaces/HoaDocument';
import sortStringDates from '../../helper-functions/sortStringDates';
import Document from '../Document/Document';

type DocumentsPageProps = {
  documents: Array<HoaDocument>;
};

function DocumentsPage({ documents }: DocumentsPageProps) {
  function displayDocuments(docs: Array<HoaDocument>) {
    return (
      <div className={classes.fileContainer}>
        <ul className={classes.fileList}>
          {docs
            .sort((x, y) => sortStringDates(x.creationDate, y.creationDate))
            .map((doc) => (
              <Document key={doc.name} document={doc} />
            ))}
        </ul>
      </div>
    );
  }

  return (
    <div className={'pageWrapper ' + classes.pageWrapperOverride}>
      <h1>Documents</h1>
      <h2>HOA Meeting Notes</h2>
      {displayDocuments(documents.filter((x) => x.type === 'meetingMinutes'))}

      <h2>Community Documents</h2>
      {displayDocuments(documents.filter((x) => x.type === 'misc'))}
    </div>
  );
}

export default DocumentsPage;
