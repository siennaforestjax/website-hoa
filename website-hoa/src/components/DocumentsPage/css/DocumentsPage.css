.pageWrapper {
  align-items: flex-start;
}

.docList {
  list-style-type: none;
}

h2 {
  font-size: 1.2rem;
}

h2:first-of-type {
  margin-top: 15px;
}

.fileContainer li {
  padding-left: calc(var(--gutter) * 2);
}

.fileContainer {
  display: block;
  margin-bottom: 25px;
}

.doc {
  padding: 5px;
}
