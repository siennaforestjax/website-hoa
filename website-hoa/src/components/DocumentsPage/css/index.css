@import './DocumentsPage.css';
@import './DocumentsPage-med.css';
