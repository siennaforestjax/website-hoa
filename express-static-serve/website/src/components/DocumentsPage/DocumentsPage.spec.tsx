import React from 'react';
import renderer from 'react-test-renderer';
import DocumentsPage from './DocumentsPage';

describe('The DocumentsPage Component', () => {
  describe('The Snapshot', () => {
    it('should render as expected', () => {
      const jsonTree = renderer
        .create(<DocumentsPage />)
        .toJSON();

      expect(jsonTree).toMatchSnapshot();
    });
  });
});
