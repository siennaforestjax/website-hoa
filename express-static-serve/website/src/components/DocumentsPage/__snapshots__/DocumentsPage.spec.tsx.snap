// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DocumentsPage Component The Snapshot should render as expected 1`] = `
<div
  className="pageWrapper"
>
  <h1>
    Documents
  </h1>
  <h2>
    HOA Meeting Notes
  </h2>
  <ul>
    <li>
      <a
        download={true}
        href="/meetingMinutes/SFHOA_Minutes_2022-05-26.docx"
        rel="nofollow noreferrer noopener"
        target="_blank"
      >
        SFHOA_Minutes_2022-05-26.docx
         (
        8kb
        )
      </a>
    </li>
    <li>
      <a
        download={true}
        href="/meetingMinutes/SFHOA_Minutes_2022-04-28.docx"
        rel="nofollow noreferrer noopener"
        target="_blank"
      >
        SFHOA_Minutes_2022-04-28.docx
         (
        10kb
        )
      </a>
    </li>
  </ul>
  <h2>
    Community Documents
  </h2>
  <ul>
    <li>
      Covenants & Restrictions
    </li>
    <li>
      Proxy Form
    </li>
    <li>
      Contact List (log in)
    </li>
  </ul>
</div>
`;
